cub/cub/util_device.cuh

```diff
@@ -53,13 +47,16 @@
 // for backward compatibility
 #include <cub/util_temporary_storage.cuh>
 
-#include <cuda/std/__cuda/ensure_current_device.h> // IWYU pragma: export
 #include <cuda/std/type_traits>
 #include <cuda/std/utility>
 
-#include <array>
-#include <atomic>
-#include <cassert>
+#if !_CCCL_COMPILER(NVRTC)
+#  include <cuda/std/__cuda/ensure_current_device.h> // IWYU pragma: export
+
+#  include <array>
```

Comment:
We should not need `<array>` and `<cassert>` we have them in `cuda/std`

Reply:
Agreed, but it's used together with `<atomic>` for a host side cache, so switching to libcu++ wouldn't gain us anything.

Reply:
we also have `<cuda/std/atomic>`

Reply:
I know, but the feature is not needed for device code in my understanding.